site/static/api/device-registry-v1.yaml

```diff
@@ -0,0 +1,724 @@
+      managed by a higher level system. In this case all calls should simply
+      return `404`. However, if the `tenants` resource is implemented, then all
+      operations of it must be implemented.
+
+      ## Security
+
+      This specification explicitly leaves out the part of authenticating and
+      authorizing users with the device registry. It is assumed that some form
+      of token exchange between the user agent and the backend service will
+      take place. Like for example HTTP basic authentication, or a bearer token.
+
+      ## Code generation
+
+      This model is not optimized for generating code from it. Code generators
+      try to understand the model and then translate this into the require
+      programming language. Even if the there would be no bugs in the code
+      generators, that process would be already only be an approximation. So
+      this model does focus in the description of the API, and doesn't tweak
+      the specification in a way to please code generators.
+
+   contact:
+      email: "hono-dev@eclipse.org"
+   license:
+      name: EPL-2.0
+      url: https://www.eclipse.org/legal/epl-2.0/
+   version: 1.0.0
+
+externalDocs:
+   description: Eclipse Hono™ homepage
+   url: https://eclipse.org/hono
+
+tags:
+   - name: tenants
+     description: Tenant Management (optional)
+     externalDocs:
+        description: Hono Multi-Tenancy
+        url: https://www.eclipse.org/hono/concepts/tenancy/
+   - name: devices
+     description: Device registration
+     externalDocs:
+        description: Hono device identity
+        url: https://www.eclipse.org/hono/concepts/device-identity/
+   - name: credentials
+     description: Device credentials
+     externalDocs:
+        description: Hono device identity
+        url: https://www.eclipse.org/hono/concepts/device-identity/
+
+servers:
+   - url: '{corsProxy}{server}/v1'
+     variables:
+        server:
+           default: http://hono.eclipse.org:28080
+        corsProxy:
+           default: https://cors-anywhere.herokuapp.com/
+
+security:
+   - BearerAuth: []
+   - BasicAuth: []
+
+paths:
+
+# Tenant API
+
+   /tenants:
+
+      post:
+         tags:
+            - tenants
+         summary: Create new tenant with auto-generated ID
+         operationId: createTenant
+         requestBody:
+            description: New tenant information
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Tenant'
+            required: true
+         responses:
+            201:
+               $ref: '#/components/responses/Created'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            409:
+               $ref: '#/components/responses/AlreadyExists'
```

Comment:
IMHO this cannot happen as the ID is auto generated, can it?

Reply:
Indeed. Fixed.